src/modules/auctions/nft-marketplace.abi.service.ts

```diff
@@ -408,16 +401,14 @@ export class NftMarketplaceAbiService {
     marketplaceAuctionId: number,
   ): Promise<TransactionNode> {
     const { collection, nonce } = getCollectionAndNonceFromIdentifier(request.identifier);
-
-    return contract
-      .call({
-        func: new ContractFunction('bid'),
-        value: TokenPayment.egldFromBigInteger(request.price),
-        args: [new U64Value(new BigNumber(marketplaceAuctionId)), BytesValue.fromUTF8(collection), BytesValue.fromHex(nonce)],
-        gasLimit: gas.bid,
-        chainID: mxConfig.chainID,
-      })
-      .toPlainObject(new Address(ownerAddress));
+    return contract.methodsExplicit
+      .bid([new U64Value(new BigNumber(marketplaceAuctionId)), BytesValue.fromUTF8(collection), BytesValue.fromHex(nonce)])
```

Comment:
Maybe the `bid` endpoint receive the `nonce` as u64? However, this still works, since "top-level encoding" is employed (and u64 and bytes are compatible in that context, up to a certain data length), and `interaction.check()` isn't called (that would have normally thrown).

https://github.com/multiversx/mx-sdk-js-core/blob/main/src/smartcontracts/interaction.ts#L229